pilot/pkg/networking/core/v1alpha3/cluster_test.go

```diff
@@ -884,6 +884,43 @@ func TestConditionallyConvertToIstioMtls(t *testing.T) {
 			},
 			autoDetected,
 		},
+		{
+			"Auto fill nil settings when mTLS nil for internal service in permissive mode",
+			nil,
+			[]string{"spiffee://foo/serviceaccount/1"},
+			"foo.com",
+			&model.Proxy{Metadata: &model.NodeMetadata{}},
+			true, false, authn_model.MTLSPermissive,
+			&networking.TLSSettings{
+				Mode:              networking.TLSSettings_ISTIO_MUTUAL,
+				CaCertificates:    constants.DefaultRootCert,
+				ClientCertificate: constants.DefaultCertChain,
+				PrivateKey:        constants.DefaultKey,
+				SubjectAltNames:   []string{"spiffee://foo/serviceaccount/1"},
+				Sni:               "foo.com",
+			},
+			autoDetected,
+		},
+		{
+			"Auto fill nil settings when mTLS nil for internal service in plaintext mode",
```

Comment:
both test case say `auto fill nil` the actual expected result is different, mind clarifying on is user supplied one is autoDetected?

Reply:
all prev auto fil imply using transport_socket_match.

so altenratively change this to do not auto fill in plaintext.